cli/pack/common.go

```diff
@@ -0,0 +1,510 @@
+	if err != nil {
+		return err
+	}
+
+	// copying application
+	err = copy.Copy(pathToCopy, filepath.Join(packagePath, filepath.Base(pathToCopy)), copy.Options{
+		Skip: func(src string) (bool, error) {
+			fileInfo, err := os.Stat(src)
+			if err != nil {
+				return false, fmt.Errorf("Failed to check if file is a socket: %s", src)
+			}
+			perm := fileInfo.Mode()
+			if perm&os.ModeSocket != 0 {
+				return true, nil
+			}
+			return false, nil
+		},
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func copyArtifacts(ctx *PackCtx, appName, packagePath string) error {
+	log.Infof("Copying all artifacts")
+
+	var (
+		baseVarDataPath = filepath.Join(packagePath, varDataPath)
+		baseVarLogPath  = filepath.Join(packagePath, varLogPath)
+		baseVarRunPath  = filepath.Join(packagePath, varRunPath)
+	)
+
+	err := copy.Copy(filepath.Join(ctx.App.DataDir, appName),
+		filepath.Join(baseVarDataPath, appName))
+	if err != nil {
+		return err
+	}
+	err = copy.Copy(filepath.Join(ctx.App.LogDir, appName),
+		filepath.Join(baseVarLogPath, appName))
+	if err != nil {
+		return err
+	}
+	err = copy.Copy(filepath.Join(ctx.App.RunDir, appName),
+		filepath.Join(baseVarRunPath, appName))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// TODO replace by tt enable
+// creates a relative link for an application that must be packed
+func createAppSymlink(ctx *PackCtx, appName, packagePath string) error {
+	appPath, err := resolveAppName(filepath.Join(ctx.App.InstancesEnabled, appName))
+	if err != nil {
+		return err
+	}
+
+	err = os.Symlink(filepath.Join("..", filepath.Base(appPath)),
+		filepath.Join(packagePath, instancesEnabledPath, appName))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func createEnv(ctx *PackCtx, destPath string) error {
+	log.Infof("Generating new tarantool.yaml for the new package")
+
+	appOpts := config.AppOpts{
+		InstancesEnabled:   instancesEnabledPath,
+		InstancesAvailable: ".",
+		BinDir:             filepath.Join(envPath, binPath),
+		RunDir:             filepath.Join(varPath, runPath),
+		DataDir:            filepath.Join(varPath, dataPath),
+		LogDir:             filepath.Join(varPath, logPath),
+		LogMaxSize:         ctx.App.LogMaxSize,
+		LogMaxAge:          ctx.App.LogMaxAge,
+		LogMaxBackups:      ctx.App.LogMaxBackups,
+		Restartable:        ctx.App.Restartable,
+	}
+	moduleOpts := config.ModulesOpts{
+		Directory: filepath.Join(envPath, modulesPath),
+	}
+	cliOptsNew := config.CliOpts{
+		App:     &appOpts,
+		Modules: &moduleOpts,
+	}
+	cfg := config.Config{
+		CliConfig: &cliOptsNew,
+	}
+
+	file, err := os.Create(filepath.Join(destPath, "tarantool.yaml"))
+	defer file.Close()
+	if err != nil {
+		return err
+	}
+
+	err = yaml.NewEncoder(file).Encode(&cfg)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func resolveAppName(appName string) (string, error) {
+	// detecting if the application is a file or a directory
+	_, err := os.Stat(appName)
```

Comment:
Is this appName or app path? If I don't mistake we are not in instances enabled dir here. How stat is going to finr app name in current directory?